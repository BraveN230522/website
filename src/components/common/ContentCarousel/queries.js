import { useQuery } from '@apollo/react-hooks';
import { get } from 'lodash';

import Loader from '@components/Loader';
import console from '@utils/console';
import { GET_NEWS } from '../../../graphql/queries/news';

const NewsQuery = ({ children, interestId, interestType, shouldRefetch, refetchInterestId, refetchInterestType }) => {
  const { loading, error, data, refetch } = useQuery(GET_NEWS, {
    variables: { interestId, interestType },
  });

  // TODO: deal with loading and error
  if (loading) {
    return <Loader />;
  }

  if (error) {
    console.error(error, 'error on league news');
    if (shouldRefetch && refetchInterestId > -1 && refetchInterestType) {
      refetch({
        interestId: refetchInterestId,
        interestType: refetchInterestType,
      });
    }
  }

  const queryData = get(data, 'getNews', []);

  if (queryData?.length === 0 && shouldRefetch && refetchInterestId > -1 && refetchInterestType) {
    refetch({
      interestId: refetchInterestId,
      interestType: refetchInterestType,
    });
  }

  return children(queryData);
};

export { NewsQuery };
