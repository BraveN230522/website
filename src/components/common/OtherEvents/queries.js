import gql from 'graphql-tag';
import { useQuery } from '@apollo/react-hooks';
import console from '@utils/console';

const GET_NEARBY_EVENTS_BY_CITY = gql`
  query nearbyEventsByCityId($cityId: Int!) {
    nearbyEventsByCityId(cityId: $cityId) {
      eventDateTime
      eventID
      eventImage
      eventName
      venue {
        venueAddress
        venueID
        venueImage
        venueName
      }
      fromPrice {
        amount
        currencyId
      }
    }
  }
`;

const GET_EVENTS_NEAR_HOTEL = gql`
  query eventsNearHotel($hotelId: String!) {
    eventsNearHotel(hotelId: $hotelId) {
      eventDateTime
      eventID
      eventImage
      eventName
      fromPrice {
        amount
        currencyId
      }
      venue {
        venueName
      }
    }
  }
`;

export const NearbyEventsByCityQuery = ({ children, cityId }) => {
  const { loading, error, data } = useQuery(GET_NEARBY_EVENTS_BY_CITY, {
    variables: { cityId },
  });

  // TODO: deal with loading and error
  if (loading) {
    console.info('Loading');
  }

  if (error) {
    console.error(error, 'error');
  }

  const nearbyCities = ((data && data.nearbyEventsByCityId) || []).sort(
    (a, b) => new Date(a.eventDateTime) - new Date(b.eventDateTime)
  );

  return children({
    nearbyCities,
  });
};

export const NearbyEventsByHotelQuery = ({ children, hotelId }) => {
  const { loading, error, data } = useQuery(GET_EVENTS_NEAR_HOTEL, {
    variables: { hotelId },
  });

  // TODO: deal with loading and error
  if (loading) {
    console.info('Loading');
  }

  if (error) {
    console.error(error, 'error');
  }

  const nearbyEvents = (data && data.eventsNearHotel) || [];

  return children({
    nearbyEvents,
  });
};

export default NearbyEventsByCityQuery;
