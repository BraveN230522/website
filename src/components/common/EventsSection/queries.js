import React from 'react';
import { useQuery } from '@apollo/react-hooks';
import Loader from '@components/Loader';

import { GET_EVENTS } from '@graphql/queries/events';
import { GET_TOP_CITIES } from '@graphql/queries/cities';

const EventsQuery = ({ children }) => {
  const { loading, error, data } = useQuery(GET_EVENTS);

  if (loading) {
    return <Loader />;
  }

  if (error) {
    const msg = `Error loading sports`;
    // here set global error state (with useContext or return a toast notification or other error component
    return <div>{msg}</div>;
  }

  if (!data.allExperiences) {
    data.allExperiences = [];
  }

  if (!data.allPopularEvents) {
    data.allPopularEvents = [];
  }

  // The data is not paginated on backend, it's keeped in FE application memory
  return children({
    ...data,
  });
};

const TopCitiesQuery = ({ children }) => {
  const { loading, error, data } = useQuery(GET_TOP_CITIES);

  if (loading) {
    // return <div>Loading...</div>;
    return null;
  }

  if (error) {
    const msg = `Error loading cities`;
    // here set global error state (with useContext or return a toast notification or other error component
    return <div>{msg}</div>;
  }

  if (!data.topCities) {
    data.topCities = [];
  }

  // The data is not paginated on backend, it's keeped in FE application memory
  return children(data.topCities);
};

export { EventsQuery, TopCitiesQuery };
