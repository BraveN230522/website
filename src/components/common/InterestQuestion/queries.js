import { useQuery } from '@apollo/react-hooks';
import { get } from 'lodash';
import { GET_INTEREST_QUESTIONS } from '../../../graphql/queries/interest-data';

const InterestQuestionsQuery = ({ children, interestId, interestType }) => {
  const { loading, error, data } = useQuery(GET_INTEREST_QUESTIONS, {
    variables: { interestId, interestType },
  });

  // TODO: deal with loading and error
  if (loading) {
    console.info('Loading');
  }

  if (error) {
    console.error(error, `Error fetching interest questions for interest ${interestType} on id ${interestId}`);
  }

  const queryData = get(data, 'interestQuestions', []);

  return children(queryData);
};

export default InterestQuestionsQuery;
