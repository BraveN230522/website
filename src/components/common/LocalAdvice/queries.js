import { useQuery } from '@apollo/react-hooks';
import { get } from 'lodash';
import console from '@utils/console';
import { GET_INTEREST_DATA } from '../../../graphql/queries/interest-data';

const InterestDataQuery = ({ children, interestId, interestType }) => {
  const { loading, error, data } = useQuery(GET_INTEREST_DATA, {
    variables: { interestId, interestType },
  });

  // TODO: deal with loading and error
  if (loading) {
    console.info('Loading');
  }

  if (error) {
    console.error(error, `Error fetching interest data for interest ${interestType} on id ${interestId}`);
  }

  const queryData = get(data, 'interestData', []);

  return children(queryData);
};

export default InterestDataQuery;
