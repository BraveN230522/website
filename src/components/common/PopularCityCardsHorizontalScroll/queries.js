import { useQuery } from '@apollo/react-hooks';
import { get } from 'lodash';
import console from '@utils/console';
import { GET_POPULAR_SPORTING_CITIES } from '../../../graphql/queries/cities';

const MostPopularSportingCitiesQuery = ({ children }) => {
  const { loading, error, data } = useQuery(GET_POPULAR_SPORTING_CITIES);

  if (loading) {
    console.info('Loading Most Popular Sporting Cities');
  }

  if (error) {
    console.error(error, `Error fetching popularSportingCities`);
  }

  const queryData = get(data, 'popularSportingCities', {});

  return children(queryData);
};

export default MostPopularSportingCitiesQuery;
