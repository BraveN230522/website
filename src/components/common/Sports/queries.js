import { GET_SPORTS } from '@graphql/queries/sports';
import { useQuery } from '@apollo/react-hooks';

const SportsQuery = ({ children }) => {
  const { loading, error, data } = useQuery(GET_SPORTS);

  if (loading) {
    return null;
    // return <div>Loading...</div>;
  }

  if (error || !data.allSports) {
    const msg = `Error loading sports`;
    // here set global error state (with useContext or return a toast notification or other error component
    return <div>{msg}</div>;
  }

  // The data is not paginated on backend, it's keeped in FE application memory
  return children({
    ...data,
  });
};

export { SportsQuery };
