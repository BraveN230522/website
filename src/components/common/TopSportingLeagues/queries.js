import { GET_TOP_LEAGUES } from '@graphql/queries/league';
import { useQuery } from '@apollo/react-hooks';

const TopLeaguesQuery = ({ children }) => {
  const { loading, error, data } = useQuery(GET_TOP_LEAGUES);

  if (loading) {
    return null;
    // return <div>Loading...</div>;
  }

  if (error) {
    const msg = `Error loading top sporting leagues`;
    // here set global error state (with useContext or return a toast notification or other error component
    return <div>{msg}</div>;
  }

  if (!data.topLeagues) {
    data.topLeagues = [];
  }

  // The data is not paginated on backend, it's keeped in FE application memory
  return children(data?.topLeagues);
};

export default TopLeaguesQuery;
