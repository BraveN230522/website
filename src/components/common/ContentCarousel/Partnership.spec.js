import React from 'react';
import { render, act, waitFor, fireEvent } from '@testing-library/react';
import withGlobals from '@swiamTestUtils/withGlobals';
import { GET_BLOG_POST } from '@graphql/queries/blog-post';
import { MockedProvider } from '@apollo/react-testing';

import Partnership from './index';

const mocks = [
  {
    request: {
      query: GET_BLOG_POST,
      variables: {},
    },
    result: {
      data: {
        allBlogPosts: Array.from({ length: 5 }, (v, k) => ({
          id: k,
          title: `Trip Ideas ${k}`,
          summary: `Atalanta B.C. fans: One of the most extreme Ultra cultures in European Football ${k}`,
          imageURL:
            'https://i0.wp.com/blog.sportswhereiam.com/wp-content/uploads/2020/01/Atalanta-BC-Ultras-Fans-Bergamo-Serie-A.jpg?w=1920&ssl=1',
          link:
            'https://blog.sportswhereiam.com/2020/01/atalanta-b-c-fans-one-of-the-most-extreme-ultra-cultures-in-european-football/',
        })),
      },
    },
  },
];

/**
 * TODO: When graphql request is configured,
 * this toHaveContent tests will be changed for a mock response
 */
describe('<Partnership />', () => {
  it('Renders the Partnership unchanged', async () => {
    let container;
    let getByText;

    act(() => {
      const component = render(
        <MockedProvider mocks={mocks} addTypename={false}>
          <Partnership />
        </MockedProvider>,
        {
          wrapper: withGlobals,
        }
      );

      container = component.container;
      getByText = component.getByText;
    });

    expect(container).toHaveTextContent('Loading');

    await waitFor(() => getByText(/Trip Ideas 0/));

    expect(container).toMatchSnapshot();
  });

  it('Changes the visible partner information', async () => {
    let container;
    let getByText;
    let getByRole;

    act(() => {
      const component = render(
        <MockedProvider mocks={mocks} addTypename={false}>
          <Partnership />
        </MockedProvider>,
        {
          wrapper: withGlobals,
        }
      );

      container = component.container;
      getByText = component.getByText;
      getByRole = component.getByRole;
    });

    expect(container).toHaveTextContent('Loading');

    await waitFor(() => getByText(/Trip Ideas 0/));

    fireEvent.click(getByRole('control-2'));

    expect(container).toHaveTextContent('Trip Ideas 1');
  });

  it('Does not change the visible partner information', async () => {
    let container;
    let getByRole;
    let getByText;

    act(() => {
      const component = render(
        <MockedProvider mocks={mocks} addTypename={false}>
          <Partnership />
        </MockedProvider>,
        {
          wrapper: withGlobals,
        }
      );

      container = component.container;
      getByRole = component.getByRole;
      getByText = component.getByText;
    });

    expect(container).toHaveTextContent('Loading');

    await waitFor(() => getByText(/Trip Ideas 0/));

    fireEvent.click(getByRole('control-1'));

    expect(container).toHaveTextContent('Trip Ideas 0');
  });
});
