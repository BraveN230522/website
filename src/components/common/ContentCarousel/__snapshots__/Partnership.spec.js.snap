// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Partnership /> Renders the Partnership unchanged 1`] = `
.emotion-33 {
  background: #2891ff;
  min-height: 256px;
  background-size: contain;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding-right: 2rem;
}

@media (max-width:60em) {
  .emotion-33 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    padding-right: 0;
  }
}

.emotion-10 {
  display: block;
  max-width: 720px;
  width: 100%;
  position: relative;
  margin: 0;
  margin-right: 37px;
  min-height: 256px;
}

@media (max-width:77.5em) {
  .emotion-10 {
    max-width: 600px;
  }
}

@media (max-width:64.375em) {
  .emotion-10 {
    max-width: 500px;
  }
}

@media (max-width:60em) {
  .emotion-10 {
    max-width: 100%;
    margin-right: 0;
    min-height: 131px;
  }
}

.emotion-0 {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  object-fit: cover;
  object-position: top;
  -webkit-transition: all ease-in 0.2s;
  transition: all ease-in 0.2s;
  opacity: 1;
}

.emotion-2 {
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  object-fit: cover;
  object-position: top;
  -webkit-transition: all ease-in 0.2s;
  transition: all ease-in 0.2s;
  opacity: 0;
}

.emotion-31 {
  margin: 49px 0;
  max-width: 521px;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

@media (max-width:60em) {
  .emotion-31 {
    max-width: 100%;
    text-align: center;
    padding: 0 50px;
    margin-top: 24px;
  }
}

.emotion-12 {
  color: #ffffff;
  margin: 0;
  font-size: 28px;
  -webkit-letter-spacing: -0.2px;
  -moz-letter-spacing: -0.2px;
  -ms-letter-spacing: -0.2px;
  letter-spacing: -0.2px;
}

.emotion-14 {
  color: #ffffff;
  font-size: 14px;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
  margin: 13px 0 16px;
}

@media (max-width:60em) {
  .emotion-14 {
    margin-bottom: 29px;
  }
}

.emotion-16 a {
  color: #ffffff;
  cursor: pointer;
  width: 100px;
  height: 39px;
  border-radius: 19.5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: 600;
  font-size: 14px;
  border: solid 2px #ffffff;
  -webkit-transition: all ease-in 0.2s;
  transition: all ease-in 0.2s;
  margin-bottom: 28px;
}

.emotion-16 a:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background: #ffffff;
  color: #2891ff;
}

@media (max-width:960px) {
  .emotion-16 a {
    width: 100%;
    height: 39px;
    font-size: 16px;
    font-weight: 600;
    margin-bottom: 32px;
    font-family: DINAlternateRegular,sans-serif;
  }
}

.emotion-29 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (max-width:60em) {
  .emotion-29 {
    margin-bottom: 32px;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

.emotion-18 {
  width: 31px;
  height: 5px;
  display: block;
  border-radius: 4px;
  background-color: #ffffff;
  margin-left: 4px;
  cursor: pointer;
  -webkit-transition: all ease-in 0.2s;
  transition: all ease-in 0.2s;
}

.emotion-18:hover {
  width: 31px;
}

.emotion-20 {
  width: 13px;
  height: 5px;
  display: block;
  border-radius: 4px;
  background-color: #53A7FF;
  margin-left: 4px;
  cursor: pointer;
  -webkit-transition: all ease-in 0.2s;
  transition: all ease-in 0.2s;
}

.emotion-20:hover {
  width: 31px;
}

<div>
  <section
    class="emotion-33 emotion-34"
  >
    <figure
      class="emotion-10 emotion-11"
    >
      <img
        class="emotion-0 emotion-1"
        src="https://i0.wp.com/blog.sportswhereiam.com/wp-content/uploads/2020/01/Atalanta-BC-Ultras-Fans-Bergamo-Serie-A.jpg?w=1920&ssl=1"
      />
      <img
        class="emotion-2 emotion-1"
        src="https://i0.wp.com/blog.sportswhereiam.com/wp-content/uploads/2020/01/Atalanta-BC-Ultras-Fans-Bergamo-Serie-A.jpg?w=1920&ssl=1"
      />
      <img
        class="emotion-2 emotion-1"
        src="https://i0.wp.com/blog.sportswhereiam.com/wp-content/uploads/2020/01/Atalanta-BC-Ultras-Fans-Bergamo-Serie-A.jpg?w=1920&ssl=1"
      />
      <img
        class="emotion-2 emotion-1"
        src="https://i0.wp.com/blog.sportswhereiam.com/wp-content/uploads/2020/01/Atalanta-BC-Ultras-Fans-Bergamo-Serie-A.jpg?w=1920&ssl=1"
      />
      <img
        class="emotion-2 emotion-1"
        src="https://i0.wp.com/blog.sportswhereiam.com/wp-content/uploads/2020/01/Atalanta-BC-Ultras-Fans-Bergamo-Serie-A.jpg?w=1920&ssl=1"
      />
    </figure>
    <article
      class="emotion-31 emotion-32"
    >
      <header>
        <h1
          class="emotion-12 emotion-13"
        >
          Trip Ideas 0
        </h1>
      </header>
      <p
        class="emotion-14 emotion-15"
      >
        Atalanta B.C. fans: One of the most extreme Ultra cultures in European Football 0
      </p>
      <div
        class="emotion-16 emotion-17"
      >
        <a
          href="https://blog.sportswhereiam.com/2020/01/atalanta-b-c-fans-one-of-the-most-extreme-ultra-cultures-in-european-football/"
        >
          More
        </a>
      </div>
      <div
        class="emotion-28 emotion-29 emotion-30"
      >
        <span
          class="emotion-18 emotion-19"
          role="control-1"
        />
        <span
          class="emotion-20 emotion-19"
          role="control-2"
        />
        <span
          class="emotion-20 emotion-19"
          role="control-3"
        />
        <span
          class="emotion-20 emotion-19"
          role="control-4"
        />
        <span
          class="emotion-20 emotion-19"
          role="control-5"
        />
      </div>
    </article>
  </section>
</div>
`;
